Dropbox/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/dropbox/DropboxAuthenticator.java

```diff
@@ -0,0 +1,261 @@
+
+    private static Log log = LogFactory.getLog(DropboxAuthenticator.class);
+
+    /**
+     * Get Dropbox authorization endpoint.
+     */
+    @Override
+    protected String getAuthorizationServerEndpoint(Map<String, String> authenticatorProperties) {
+        return DropboxAuthenticatorConstants.Dropbox_OAUTH_ENDPOINT;
+    }
+
+    /**
+     * Get Dropbox token endpoint.
+     */
+    @Override
+    protected String getTokenEndpoint(Map<String, String> authenticatorProperties) {
+        return DropboxAuthenticatorConstants.Dropbox_TOKEN_ENDPOINT;
+    }
+
+    /**
+     * Get Dropbox user info endpoint.
+     */
+    @Override
+    protected String getUserInfoEndpoint(OAuthClientResponse token, Map<String, String> authenticatorProperties) {
+        return DropboxAuthenticatorConstants.Dropbox_USERINFO_ENDPOINT;
+    }
+
+    /**
+     * Check ID token in Dropbox OAuth.
+     */
+    @Override
+    protected boolean requiredIDToken(Map<String, String> authenticatorProperties) {
+        return false;
+    }
+
+    /**
+     * Get the friendly name of the Authenticator
+     */
+    @Override
+    public String getFriendlyName() {
+        return DropboxAuthenticatorConstants.AUTHENTICATOR_FRIENDLY_NAME;
+    }
+
+    /**
+     * Get the name of the Authenticator
+     */
+    @Override
+    public String getName() {
+        return DropboxAuthenticatorConstants.AUTHENTICATOR_NAME;
+    }
+
+    /**
+     * Get Configuration Properties
+     */
+    @Override
+    public List<Property> getConfigurationProperties() {
+        List<Property> configProperties = new ArrayList<Property>();
+        Property clientId = new Property();
+        clientId.setName(OIDCAuthenticatorConstants.CLIENT_ID);
+        clientId.setDisplayName(DropboxAuthenticatorConstants.CLIENT_ID);
+        clientId.setRequired(true);
+        clientId.setDescription("Enter Foursquare client identifier value");
+        clientId.setDisplayOrder(0);
+        configProperties.add(clientId);
+
+        Property clientSecret = new Property();
+        clientSecret.setName(OIDCAuthenticatorConstants.CLIENT_SECRET);
+        clientSecret.setDisplayName(DropboxAuthenticatorConstants.CLIENT_SECRET);
+        clientSecret.setRequired(true);
+        clientSecret.setConfidential(true);
+        clientSecret.setDescription("Enter Foursquare client secret value");
+        clientSecret.setDisplayOrder(1);
+        configProperties.add(clientSecret);
+
+        Property callbackUrl = new Property();
+        callbackUrl.setDisplayName(DropboxAuthenticatorConstants.CALLBACK_URL);
+        callbackUrl.setName(IdentityApplicationConstants.OAuth2.CALLBACK_URL);
+        callbackUrl.setDescription("Enter the callback url");
+        callbackUrl.setDisplayOrder(2);
+        configProperties.add(callbackUrl);
+        return configProperties;
+    }
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
```

Comment:
Do we need to override processAuthenticationResponse here? Can't we directly use the method from super class?